cdk/stack.py

```diff
@@ -98,6 +102,76 @@ def __init__(
             description="Name of table used to store ingestions",
         )
 
+        env_secret = self.build_env_secret(config.stage, env)
+        secret_arn: str = env_secret.secret_arn
+
+        self.build_oidc(
```

Comment:
can we move the `if oidc_provider_arn:` condition here instead and remove it from `build_oidc`?